src/rw.jl

```diff
@@ -0,0 +1,196 @@
+        end
+
+    end
+
+    """
+    Encodes the loop condition for the reading step.
+    """
+
+    function reading(; timeout = 100.0)
+
+        loop_condition = begin_step(engine, step_mode_read, timeout) != step_status_end_of_stream
+
+        if loop_condition
+            global reading_now = true
+            return true
+        else
+            global reading_now = false
+            global nprocessed = 0
+            return false
+        end
+        
+    end
+
+    """
+    Reads the data sequentially. This function is particularly useful when called from a loop.
+    """
+
+    function read(var; verbose = true)
+
+        var_id = inquire_variable(io, var)
+
+        if nprocessed == 0
+            nxy_global = shape(var_id)                                               # Extract meta data
+            nxy        = count(var_id)                                               # ...
+            var_type   = type(var_id)                                                # ...
+            global V   = zeros(var_type, nxy)                                        # Preallocate memory for the variable using the meta data
+        end
+
+        get(engine, var_id, V)
+
+        if reading                                                                  # end the step automatically if loop detected
+            end_step(engine)
+            global nprocessed += 1
+        end
+
+        if verbose
+            print("Variable: " * string(var))
+            print("Step: " * string(nprocessed))
+            print("Variable steps: " * string(steps(var_iq)))
+            print("Current step: " * string(current_step(engine)))
+        end
+
+    end
+
+    """
+    Provide details about all avaiable variables listed in the ADIOS2 file.
+    """
+
+    function inspect_variables(filename::AbstractString = "")
+        if !isempty(filename)
+            var_info = adios_all_variable_names(filename)
+            print(var_info)
+        else
+            print("error")
+        end
+    end
+
+    """
+    Close the ADIOS2 engine once all steps have been performed.
+    """
+    function finalize_adios()
+
+        if init_state == true
+            close(engine)
+        else
+            print("Error. ADIOS2 hasn't been initialized.")
+        end
+
+    end
+
+    """
+    If the ADIOS2 config file is not provided, this function uses the XML
+    template to create the file automatically with parameters provided by the
+    user. Returns the XML path to the new file.
+    """
+    function adios2_config(engine_type = "")
+
+        template_path = joinpath(pwd(), "adios2_template.xml")
+        xml_path = joinpath(pwd(), "adios2_config.xml") # save location
+
+        adios2_template = open(template_path, "r")
+
+        adios2_xml = open(xml_path, "w")
+
+        cp(template_path, xml_path, force = true)
```

Comment:
Here we should not copy the file, but read the whole template file at once into memory as string, do the required modifications, and then write the whole config file at once.